src/aiai_eval/task_factory.py

```diff
@@ -20,14 +24,40 @@ class TaskFactory:
     def __init__(self, evaluation_config: EvaluationConfig):
         self.evaluation_config = evaluation_config
 
-    def build_dataset(self, dataset_task: Union[str, DatasetTask]) -> list:
+    def build_dataset_task(self, dataset: Union[str, DatasetTask]) -> EvaluationDataset:
```

Comment:
If we stick to tasks, shouldn't this be a task rather than a dataset? Same counts throughout the other scripts.

Reply:
I agree see below.